src/SimpleSAML/Configuration.php

```diff
@@ -558,6 +559,25 @@ public function getPathValue(string $name, ?string $default = null): ?string
     }
 
 
+    /**
+     * Retrieve the location of the vendor directory
+     *
+     * This function checks whether SimpleSAMLphp is installed as a stand-alone application or as a library
+     * and determines the location of the vendor directory.
+     *
+     * @return string The absolute path to the vendor directory. This path will always end with a slash.
+     */
+    public function getVendorDir(): string
+    {
+        if (file_exists(dirname(__FILE__, 3) . '/vendor')) {
+            return dirname(__FILE__, 3) . '/vendor/';
+        } else {
+            // SSP is loaded as a library.
+            return dirname(__FILE__, 3) . '/../../vendor/';
```

Comment:
I think we need to go one more level higher.

I tried this test:
```
(new XML)->isValid('<root></root>', 'saml-schema-metadata-2.0.xsd');
```

And got this PHP warning:
```
PHP Warning:  DOMDocument::schemaValidate(): Invalid Schema in /app/test/vendor/simplesamlphp/simplesamlphp/src/SimpleSAML/Utils/XML.php on line 364
PHP Stack trace:
PHP   1. {main}() /app/test/test.php:0
PHP   2. SimpleSAML\Utils\XML->isValid($xml = '<root></root>', $schema = 'saml-schema-metadata-2.0.xsd') /app/test/test.php:8
PHP   3. DOMDocument->schemaValidate($filename = '/app/test/vendor/simplesamlphp/simplesamlphp/../../vendor/simplesamlphp/saml2/resources/schemas/saml-schema-metadata-2.0.xsd') /app/test/vendor/simplesamlphp/simplesamlphp/src/SimpleSAML/Utils/XML.php:364
Schema validation failed on XML string:
level=1,code=1549,line=0,col=0,msg=failed to load external entity "/app/test/vendor/vendor/simplesamlphp/saml2/resources/schemas/saml-schema-metadata-2.0.xsd"
level=2,code=1757,line=0,col=0,msg=Failed to locate the main schema resource at '/app/test/vendor/vendor/simplesamlphp/saml2/resources/schemas/saml-schema-metadata-2.0.xsd'.
```

```suggestion
            return dirname(__FILE__, 3) . '/../../../vendor/';
```